README.md
JcUtils
=======

A personal collection of Perl utilities

JcUtils/Changes
Revision history for JcUtils-Logger

0.01    Date/time
        First version, released on an unsuspecting world.



JcUtils/scripts/test.pl
#!/usr/bin/perl -w

use strict;
use JcUtils::Logger;
use 5.006;
use warnings;

my $mailLogger = JcUtils::Logger::new('/tmp/testLogger');

try($mailLogger);

sub try{
	my $logger = shift;
	
	$logger->log("some log");
}



JcUtils/blib/lib/JcUtils/Logger.pm
package JcUtils::Logger;

use 5.006;
use strict;
use warnings;

use Time::localtime;
use Time::Local;
use enum qw(WARN INFO ERROR);

my %defaults = (
	'logFile' => "/tmp/defaultLog",
	'maxLogSize' => 100000,
);

=head1 NAME

JcUtils::Logger - The great new JcUtils::Logger!

=head1 VERSION

Version 0.01

=cut

our $VERSION = '0.01';


=head1 SYNOPSIS

Perl logger utility for use in perl scripts.  Features three logging levels WARN, INFO, and ERROR.
JcUtils::Logger will never die from error; a failed log attempt should not stop a program from running.
Ability to set log level such that after debugging you can specify to log only ERROR or ERROR and WARN thus not logging INFO.

    my $defaultLogger = JcUtils::Logger::new();
    $defaultLogger->log("some text to log");
    $defaultLogger->warn->log("Warning log");
    $defaultLogger->error->log("Error log");
    $defaultLogger->log("Back to INFO log");
	
    my $newLogger = JcUtils::Logger::new("/tmp/newtestlog");
	
    my $filesizeLogger = JcUtils::Logger::new("/tmp/testLog", 10000);

=cut


=head1 EXPORT

None.

=head1 SUBROUTINES/METHODS

=cut


=head2 new()

Creates the logger object

=head2 Arguments

=over 3

=item 1. String, log file name

=item 2. Number, log file max size

=back

=cut

sub new {
	
	my $self = bless {};
	my %args;
	$self->{logEntryNum} = 0;
	$self->{logLevel} = INFO;
	
	($args{logFile}, $args{maxLogSize} ) = @_;
	
	#Has the user defined a specific log file
	if (defined($args{logFile})){
		$self->{logFile} = $args{logFile};
	}
	else {
		$self->{logFile} = $defaults{logFile};
	}
	
	#Has the user provided a max log size
	if (defined($args{maxLogSize})) {
		$self->{maxLogSize} = $args{maxLogSize};
	}
	else {
		$self->{maxLogSize} = $defaults{maxLogSize};
	}
	
	$self->{logType} = INFO;
	
	#before we open the log, let's do some maintenance
	$self->maint();
	
	$self->openLog();
	
	return $self;
}

=head2 log()

Formats strings for logging then logs string.

=over 2

=item 1. String, string to be logged

=back

=cut

sub log {
	my ($self, $entry) = @_;
	
	my $log_fh = $self->{log_fh};
	
	my $timeStamp;
	my $build;
	
	$timeStamp = localtime()->mon()+1 . localtime()->mday() . localtime()->hour() . localtime()->min() . localtime()->sec();
	
	if ($self->{logType} == INFO) {
		$build = "INFO ";
	}
	if ($self->{logType} == WARN){
		$build = "WARN "
	}
	if ($self->{logType} == ERROR){
		$build = "ERROR "
	}
	
	$build = $build . $self->{logEntryNum}++ . $timeStamp . " " . $entry . "\n";
	
	#check to see if, somehow the log was deleted
	my $logFile = $self->{logFile};
	unless (-e $logFile) {
		$self->openLog();
	}
	
	#Are we logging this entry
	if ($self->{logLevel} == ERROR) {
		if ($self->{logType} == ERROR) {
			print $log_fh $build;
			$self->{logType} = INFO;
			return 1;
		}
	}
	if ($self->{logLevel} == WARN) {
		if ($self->{logType} == WARN || $self->{logType} == ERROR ) {
			print $log_fh $build;
			$self->{logType} = INFO;
			return 1;
		}
	}
	
	if ($self->{logLevel} == INFO) {
		if ($self->{logType} == WARN || $self->{logType} == INFO || $self->{logType} == ERROR ) {
			print $log_fh $build;
			$self->{logType} = INFO;
			return 1;
		}
	}

	$self->{logType} = INFO;
	
	return 1;
	
}

=head2 setLogLevelError()

Sets logger to log only ERROR

=cut

sub setLogLevelError {
	my $self = shift;
	$self->{logLevel} = ERROR;
	return $self;
}

=head2 setLogLevelWarn()

Sets logger to log only WARN and ERROR

=cut

sub setLogLevelWarn {
	my $self = shift;
	$self->{logLevel} = WARN;
	return $self;
}

=head2 setLogLevelInfo()

Sets logger to log INFO, WARN, and ERROR.  Or, log everything as is the default.

=cut

sub setLogLevelInfo {
	my $self = shift;
	$self->{logLevel} = INFO;
	return $self;
}

=head2 info()

Sets the log entry type to INFO, returns itself

=cut

sub info {
	my $self = shift;
	$self->{logType} = INFO;
	return $self;
}

=head2 warn()

Sets the log entry type to WARN, returns itself

=cut

sub warn {
	my $self = shift;
	$self->{logType} = WARN;
	return $self;
}

=head2 error()

Sets the log entry type to ERROR, returns itself

=cut

sub error {
	my $self = shift;
	$self->{logType} = ERROR;
	return $self;
}
=head2 maint()

Does log maintenace and is usually called in the constuctor of this object; however, can be
called by the user.

=cut

sub maint {
	my $self = shift;
	my $logFile = $self->{logFile};
	my $maxLogSize = $self->{maxLogSize};

	my $size = -s $logFile;
	if (defined($size)) {
		if ($size > $maxLogSize) {
			if (-e $logFile . ".bak") {
				unlink($logFile . ".bak");
			}
			rename($logFile, $logFile . ".bak");
		}
	}
	
}

=head2 closeLog()

Closes the log file

=cut

sub closeLog {
	
	my $self = shift;
	
	my $log_fh = $self->{log_fh};

	unless (close ($log_fh)) {
		print STDERR "Unable to close log file: " . $self->{logFile} . $! . "\n";
		return 0;
	}
	return 1;
}

sub openLog {
	
	my $self = shift;
	
	unless (open ($self->{log_fh}, ">>$self->{logFile}")) {
		print STDERR "Unable to open log file: " . $self->{logFile} . $! . "\n";
		return 0;
	}
	
	#enable autoflush on the logger
	#my $log_fh = $self->{log_fh};
	#$log_fh->autoflush(1);
	$self->{log_fh}->autoflush(1);
	
	return 1;
	
}


=head1 AUTHOR

Jamie Cyr, C<< <jjcyr at yahoo.com> >>

=head1 BUGS

Please report any bugs or feature requests to TBD.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc JcUtils::Logger

You can also look for information at: TBD

=head1 ACKNOWLEDGEMENTS

=head1 LICENSE AND COPYRIGHT

Copyright 2012 Jamie Cyr.

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.


=cut

1; # End of JcUtils::Logger


JcUtils/blib/lib/JcUtils/SendGmail.pm
package JcUtils::SendGmail;

use 5.006;
use strict;
use warnings;
use JcUtils::Logger;
use Net::SMTP::TLS;

=head1 NAME

JcUtils::SendGmail - The great new JcUtils::SendGmail!

=head1 VERSION

Version 0.01

=cut

our $VERSION = '0.01';


=head1 SYNOPSIS

Quick summary of what the module does.

Perhaps a little code snippet.

    use JcUtils::SendGmail;

    my $foo = JcUtils::SendGmail->new();
    ...

=head1 EXPORT

A list of functions that can be exported.  You can delete this section
if you don't export anything, such as for a purely object-oriented module.

=cut

our %defaults = (
	'server' => "smtp.gmail.com",
	'portNumber' => '587',
	'timeOut' => 5,
	'logFile' => '/tmp/sendGmailLog',
);


=head1 SUBROUTINES/METHODS

=head2 new()

Creates the SendGmail object

=over 7

=item 1. JcUtils::Logger, logger object

=item 2. String, Gmail user name

=item 3. String, Gmail user password

=item 4. Number, server

=item 5. String, port Number

=item 6. String, time out

=back

=head2 returns

=over 2

=item 1. Object, self

=back

=cut

sub new {

	my $self = bless {};
	my $logger;
	my $args;
	
	#did they pas a hashref
	if (ref $_[0] eq 'HASH') {
		$args = shift;
		$self->{logger} = $args->{logger};
		$logger = $self->{logger};
	}
	else {
		$args = {};
		#is the first argument a logger
		my $obj = $_[0];
		my $tmp = ref($obj);
		if ($tmp eq 'JcUtils::Logger') {
			($self->{logger}, $args->{userName}, $args->{password}, $args->{server}, $args->{portNumber}, $args->{timeOut} ) = @_;
			$logger = $self->{logger};
		}
		else {
			($args->{userName}, $args->{password}, $args->{server}, $args->{portNumber}, $args->{timeOut} ) = @_;
		}
	}
	
	#do we need to create a defatul logger
	if (!defined($logger)){
		$self->{logger} = JcUtils::Logger::new($defaults{logFile});
		$logger = $self->{logger};
		$logger->log("Created default logger");
	}
	else {
		$logger->log("User provided logger");
	}
	
	if (defined($args->{server})){
		$self->{server} = $args->{server};
		$logger->log("Using user provided server: " . $self->{server});
	}
	else {
		$self->{server} = $defaults{server};
		$logger->log("Using default server: " . $self->{server});
	}
	
	if (defined($args->{portNumber})){
		$self->{portNumber} = $args->{portNumber};
		$logger->log("Using user provided portNumber: " . $self->{portNumber});
	}
	else {
		$self->{portNumber} = $defaults{portNumber};
		$logger->log("Using default port: " . $self->{portNumber});
	}
	
	if (defined($args->{timeOut})){
		$self->{timeOut} = $args->{timeOut};
		$logger->log("Using user defined timeOut: " . $self->{timeOut});
	}
	else {
		$self->{timeOut} = $defaults{timeOut};
		$logger->log("Using default timeOut: " . $self->{timeOut});
	}
	
	if (defined($args->{userName})){
		$self->{userName} = $args->{userName};
	}
	else {
		$logger->error->log("Missing necessary argument userName");
		die "Missing necessary argument userName \n";
	}
	
	if (defined($args->{password})){
		$self->{password} = $args->{password};
	}
	else {
		$logger->error->log("Missing necessary argument password");
		die "Missing necessary argument password \n";
	}
	
	$logger->log("SendGmail Ready to send");
	return $self
	
}

=head2 sendMessage()

Sends a message using gmail smtp server

=over 4

=item 1. String, To: email address

=item 2. String, message

=item 3. String, subject

=back

=head2 returns

=over 3

=item 1. Number, 0 failure

=item 2. Number, 1 success

=back

=cut

sub sendMessage {
	my $self = shift;
	my $logger;
	my $args;
	
	$logger = $self->{logger};
	
	if (ref $_[0] eq 'HASH') {
		$args = shift;
	}
	else {
		$args = {};
		($args->{to}, $args->{message}, $args->{subject}) = @_;
	}
	
	#where the right arguments supplied
	if (!defined($args->{to})) {
		$logger->error->log("To: not provided");
		return 0;
	}
	
	if (!defined($args->{message})) {
		$logger->error->log("Messge not provided");
		return 0;
	}
	
	if ($self->{userName} eq 'Fake.User@gmail.com') {
		$logger->warn->log("not actually sending the message");
		$logger->log("Sent fake message To: " . $args->{to} . " Subject: " . $args->{subject});
		return 1;
	}
	
	
	eval {
		my $mailer = new Net::SMTP::TLS(
        $self->{server},
        Port    =>      $self->{portNumber},
        User    =>      $self->{userName},
        Password=>      $self->{password});
	 $mailer->mail($self->{userName});
	 $mailer->to($args->{to});
	 $mailer->data;
	 if (defined($args->{subject})){
	 	$mailer->datasend("Subject: " . $args->{subject} . "\n");
	 }
	 else {
	 	$logger->warn->log("No subject given");
	 }
	 $mailer->datasend($args->{message} . "\n");
	 $mailer->dataend;
	 $mailer->quit;
	};
	
	if ($@) {
		$logger->error->log("Net::SMTP::TLS encountered and error $@");
		return 0;
	}
	
	$logger->log("Sent message To: " . $args->{to} . " Subject: " . $args->{subject});
	 return 1;
	
}

=head1 AUTHOR

Jamie Cyr, C<< <jjcyr at yahoo.com> >>

=head1 BUGS

Please report any bugs or feature requests to C<bug-jcutils-logger at rt.cpan.org>, or through
the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=JcUtils-Logger>.  I will be notified, and then you'll
automatically be notified of progress on your bug as I make changes.




=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc JcUtils::SendGmail


You can also look for information at:

=over 4

=item * RT: CPAN's request tracker (report bugs here)

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=JcUtils-Logger>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/JcUtils-Logger>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/JcUtils-Logger>

=item * Search CPAN

L<http://search.cpan.org/dist/JcUtils-Logger/>

=back


=head1 ACKNOWLEDGEMENTS


=head1 LICENSE AND COPYRIGHT

Copyright 2012 Jamie Cyr.

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.


=cut

1; # End of JcUtils::SendGmail


JcUtils/lib/JcUtils/FileDB.pm
package JcUtils::FileDB;

use 5.006;
use strict;
use warnings;
use JcUtils::Logger;

use Time::localtime;
use Time::Local;
use JSON::XS;
use enum qw(OPEN CLOSED);

=head1 NAME

JcUtils::FileDB - The great new JcUtils::FileDB!

=head1 VERSION

Version 0.01

=cut

our $VERSION = '0.01';


=head1 SYNOPSIS

Flat file data base for hashes.
Stores data in a flat file using JSON format.
Smartly opens and closes the flat file db; creating 10 entries in a row opens the file once and closes
it when necessary for a reading operation such at fetch();

Perhaps a little code snippet.

    my $defaultdb = JcUtils::FileDB::new();
    
    my $logger = JcUtils::Logger::new("/tmp/myLogfile");
    my $mydb = JcUtils::FileDB::new($logger, "/tmp/myDbFile");
    
    my $hashDb = JcUtils::FileDB::new({
         'dbFile'	=>	'/tmp/hashDbFile',
         'logger'	=>	$logger,
         'maxRecords'	=> 10000
    });

=head1 EXPORT

None.

=head1 SUBROUTINES/METHODS

=cut

our %defaults = (
	'dbFile' => "/tmp/defaultDbFile",
	'maxRecords' => 1000,
	'dbLogFile' => "/tmp/defaultDbFileLog",
);


=head2 new()

Cretes the new FileDb object.  You can use a hasref or individule arguments in the correct order.

=head3 Arguments

=over 4

=item 1. Object, JcUtils::Logger, user provided logger

=item 2. String, dbFile file name

=item 3. Number, maxRecords size, the max amount of records in the DB

=back

=head3 Return

=over 2

=item 1. Object, new FileDb object

=back

=cut

sub new {
	my $self = bless {};
	my $args;
	my $logger;
	
	#did they pas a hashref
	if (ref $_[0] eq 'HASH') {
		$args = shift;
		$self->{logger} = $args->{logger};
		$logger = $self->{logger};
	}
	else {
		$args = {};
		#is the first argument a logger
		my $obj = $_[0];
		my $tmp = ref($obj);
		if ($tmp eq 'JcUtils::Logger') {
			($self->{logger}, $args->{dbFile}, $args->{maxRecords}  ) = @_;
			$logger = $self->{logger};
		}
		else {
			($args->{dbFile}, $args->{maxRecords}  ) = @_;
		}
	}
	
	#do we need to create a defatult logger
	if (!defined($logger)){
		$self->{logger} = JcUtils::Logger::new($defaults{dbLogFile});
		$logger = $self->{logger};
		$logger->log("Created default logger");
	}
	else {
		$logger->log("User provided logger");
	}
	
	if (defined($args->{dbFile})){
		$self->{dbFile} = $args->{dbFile};
	}
	else {
		$self->{dbFile} = $defaults{dbFile};
	}
	
	if (defined($args->{maxRecords})) {
		$self->{maxRecords} = $args->{maxRecords};
	}
	else {
		$self->{maxRecords} = $defaults{maxRecords};
	}
	
	$self->{entryNum} = 0;
	$self->{dbState} = CLOSED;
	
	$self->openDb();
	
	$self->{logger}->log("FileDb Ready");
	
	return $self;
}

=head2 create()

Create a database record. Returns the UUID of the recored created.

=head3 Arguments

=over 2

=item 1. Hashref, hashref of the data structure to store

=back

=head3 Return

=over 3

=item 1. Number, UUID of the entry.

=item 2. Number, 0 on failure.

=back

=cut

sub create {
	my ($self, $entry) = @_;
	my $line = "";
	my $uuid;
	my $logger = $self->{logger};
	
	#We need the enty argument
	if (!defined($entry)) {
		$logger->error->log("entry argument was not defined");
		return 0;
	}
	
	#create a unique DB ID
	$uuid = $self->{entryNum}++ . localtime()->mday() . localtime()->hour() . localtime()->min() . localtime()->sec();
	$entry->{'UUID'} = $uuid;
	
	$line = JSON::XS->new->encode($entry);
	$line = $line . "\n";
	
	if ($self->{dbState} == CLOSED) {
		$self->openDb();
	}
	
	my $db_fh = $self->{db_fh};
	
	print $db_fh $line;
	$logger->log("Created entry $uuid");
	
	return $uuid;

}

=head2 update()

Udate a record in the DB

=head3 Arguments

=over 2

=item 1. Hashref, Hash reference with UUID field.

=back

=head3 Return

=over 3

=item 1. Number, 1 for Success.

=item 2. Number, 0 for failure.

=back

=cut

sub update {
	
	my ($self, $entry) = @_;
	my $db_fh;
	my $dbW_fh;
	my $line = "";
	my $dbFile = $self->{dbFile};
	my $logger = $self->{logger};
	my $record = {};
	
	#We need the enty argument
	if (!defined($entry)) {
		$logger->error->log("Entry argument was not provided");
		return 0;
	}
	
	#The entry needs a uuid
	if (!defined($entry->{UUID})) {
		$logger->error->log("Entry had no UUID");
		return 0;
	}
	
	#Make sure the filehandle is closed.
	if ($self->{dbState} == OPEN){
		$self->closeDb();
	}
	
	open ($db_fh, "<$dbFile") || die "Could not open logfile: $self->{dbFile}: $!";
	
	#Open the temp DB
	unless (open ($dbW_fh, ">>$dbFile.tmp")) {
		$logger->error->log("Failed to open tmp DB");
		return 0;
	}
	
	while (<$db_fh>) {
		chomp $_;
		$record = JSON::XS->new->decode($_);
		if (exists $record->{UUID}) {
			if ($record->{UUID} eq $entry->{UUID}) {
				$line = JSON::XS->new->encode($entry);
				$line = $line . "\n";
				print $dbW_fh $line;
				$logger->log("updated entry $entry->{UUID}");
			}
			else {
				print $dbW_fh $_ . "\n";
			}
		}
	}
	
	close ($db_fh);
	close ($dbW_fh);
	
	#rename the DB
	unlink ($dbFile) || die "Could not delete $dbFile: $! \n";
	rename ($dbFile . ".tmp", $dbFile) || die "Could not rename $dbFile.tmp: $! \n";
	
	return 1;
	
}

sub delete {
	
}

=head2 find()

Find a record in the db, return an array of UUIDs, empty array if nothing was found.  Use fetch() method to
acutally obtain the record.

=head3 Arguments

=over 4

=item 1. String, key to find

=item 2. String, the string to search for

=item 3. Number, 1=ignore case

=back

=head3 Return

=over 2

=item 1. Array, array of restuls, empty array if no results found or failure.

=back

=cut

sub find {
	
	my ($self, $key, $searchString, $ignoreCase) = @_;
	my $db_fh;
	my $dbFile = $self->{dbFile};
	my $logger = $self->{logger};
	my @results;
	my $record = {};
	
	#We need at least the first two arguments
	if (!defined($key)) {
		$logger->error->log("Key String was not defined");
		return @results;
	}
	
	if (!defined($searchString)) {
		$logger->error->log("Search String was not defined");
		return @results;
	}
	
	#Make sure the filehandle is closed.
	if ($self->{dbState} == OPEN){
		$self->closeDb();
	}
	
	open ($db_fh, "<$dbFile") || die "Could not open logfile: $self->{dbFile}: $!";
	
	$logger->log("Looking for $key, $searchString");
	
	while (<$db_fh>) {
		chomp $_;
		$record = JSON::XS->new->decode($_);
		if (exists $record->{$key}) {
			if ($record->{$key} eq $searchString) {
				$logger->log("found entry $record->{UUID}");
				push @results, $record->{UUID};
			}
		}
	}
	
	my $size = @results;
	
	unless ($size > 0){
		$logger->warn->log ("Did not find $key, $searchString");
	}
	
	close ($db_fh);
	
	return @results;
	
}

=head2 fetch()

Fetch a record from the db.  If it doesn't exist return empty hashref

=head3 Arguments

=over 2

=item 1. Number, the UUID of the record

=back

=head3 Return

=over 2

=item 1. Hashref, hash reference of the object.

=item 2. Number, 0 on failure.

=back

=cut

sub fetch {
	my ($self, $entry) = @_;
	my $db_fh;
	my $dbFile = $self->{dbFile};
	my $logger = $self->{logger};
	my $record = {};
	
	#We need the UUID argument
	if (!defined($entry)) {
		$logger->error->log("UUID argument was not provided");
		return 0;
	}
	
	#Make sure the filehandle is closed.
	if ($self->{dbState} == OPEN){
		$self->closeDb();
	}
	
	
	open ($db_fh, "<$dbFile") || die "Could not open logfile: $self->{dbFile}: $!";
	
	$logger->log("Looking for $entry");
	
	while (<$db_fh>) {
		chomp $_;
		$record = JSON::XS->new->decode($_);
		if (exists $record->{UUID}) {
			if ($record->{UUID} eq $entry) {
				$logger->log("found entry $record->{UUID}");
				return $record;
			}
		}
	}
	$logger->warn->log("Did not find $entry");
	
	close ($db_fh);
	
	return $record = {};
	
}

=head2 getRecordCount()

Get the number of records in the DB.

=head3 Returns

=over 2

=item 1. Number, number of records in the data base

=back

=cut

sub recordCount {
	my $self = shift;
	my $db_fh;
	my $dbFile = $self->{dbFile};
	my $logger = $self->{logger};
	my $count = 0;
	
	#Make sure the filehandle is closed.
	if ($self->{dbState} == OPEN){
		$self->closeDb();
	}
	
	
	open ($db_fh, "<$dbFile") || die "Could not open logfile: $self->{dbFile}: $!";
	
	$logger->log("Counting records");
	
	while (<$db_fh>) {
		$count++;
	}
	
	close ($db_fh);
	
	return $count;
	
}

=head2 closeDb()

Convenience function that opens the flat file data base.

=head3 Return

=over 3

=item 1. Number, 1 success.

=item 2. Number, 0 failure.

=back

=cut

sub closeDb {
	my $self = shift;
	my $db_fh = $self->{db_fh};
	my $logger = $self->{logger};
	
	unless (close ($db_fh)) {
		return 0;
	}
	$self->{dbState} = CLOSED;
	$logger->log("Database CLOSED");
	return 1;
}

=head2 openDb()

Convenience function that closes the flat file data base.

=head3 Return

=over 3

=item 1. Number, 1 success.

=item 2. Number, 0 failure.

=back

=cut

sub openDb {
	my $self = shift;
	my $logger = $self->{logger};
	
	my $dbFile = $self->{dbFile};
	
	unless (open ($self->{db_fh}, ">>$dbFile")) {
		$logger->error->log("Failed to open DB");
		return 0;
	}
	
	#enable autoflush on the db
	my $db_fh = $self->{db_fh};
	$db_fh->autoflush(1);
	
	$self->{dbState} = OPEN;
	$logger->log("Database OPEN");
	
	return 1;
}

=head1 AUTHOR

Jamie Cyr, C<< <jjcyr at yahoo.com> >>

=head1 BUGS

=head2 Ignore Case

The ignore case option on find() is not implemented.

=head2 delete()

The delete() function is not yet implemented.

Please report any bugs or feature requests to TBD.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc JcUtils::FileDB

You can also look for information at: TBD

=head1 ACKNOWLEDGEMENTS

None.

=head1 LICENSE AND COPYRIGHT

Copyright 2012 Jamie Cyr.

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.

=cut

1; # End of JcUtils::FileDB
